### PR TITLE
Feat/ab#60047 link to maps on geospatial fields

### PR DESCRIPTION
**Pull request by MwanPygmay**
# Description

Adds a modal for geospatial field. Each geospatial field now has a 'Display map' button. Upon clicking on it, it opens a modal zoomed on the marker. Each marker associated to the geospatial question appear on the map.

/!\ Had to invert the coordinates as they seem to be stored in the reverse order in the database. (longitude, latitude). Maybe it would need to change the saving in the database and remove my code to inverse. For now, code is functional as such.

## Ticket
https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/60047/

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (refactor or addition to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Created a form with a geospatial form. Answered it a couple of times with nearby and far-away points. created a grid with a field related to the geospatial question. Clicking on one of the buttons opens a map centered on the clicked marker. The map can be zoomed, unzoomed. All markers appear on it.

## Sreenshots

image
image

# Checklist:

( * == Mandatory ) 

- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/safe/src/lib/components/ui/core-grid/grid/grid.component.html [hunk 1]
```diff
@@ -617,6 +617,7 @@
             {{
               applyFieldFormat(getPropertyValue(dataItem, field.name), field)
             }}
+            <button kendoButton id="testButton" (click)="onOpenPopup(dataItem, field)"> Display map </button>
```
Comment (AntoineRelief) [inline libs/safe/src/lib/components/ui/core-grid/grid/grid.component.html]: @MwanPygmay either translate or mark as todo(translate) 👍

File: libs/safe/src/lib/components/ui/core-grid/map-modal/map-modal.component.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ */
+@Component({
+  selector: 'safe-map-modal',
+  templateUrl: './map-modal.component.html',
+  styleUrls: ['./map-modal.component.scss'],
+})
+export class MapModalComponent implements OnInit {
+  /**
+   * default constructor for the modal
```
Comment (AntoineRelief) [inline libs/safe/src/lib/components/ui/core-grid/map-modal/map-modal.component.ts]: @MwanPygmay avoid this type of comment
meaning that describing constructor as `constructor` is not good
we should there indicate what the component is about
this is what you will see in your IDE, when hovering component used in other component, and so, it should help to understand what the component is about

File: libs/safe/src/lib/components/ui/core-grid/map-modal/map-modal.component.scss [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.map {
```
Comment (AntoineRelief) [inline libs/safe/src/lib/components/ui/core-grid/map-modal/map-modal.component.scss]: @MwanPygmay better use tailwind for this kind of styling

Comment (MwanPygmay) [reply libs/safe/src/lib/components/ui/core-grid/map-modal/map-modal.component.scss]: @AntoineRelief  Made the changes

**Pull request merged**